### ISSUE TITLE
Determine how incorrect prop names can be caught in Angular apps

### ISSUE THREAD

**Issue by GCHQ-Developer-847** [mi6/ic-design-system#499]
## Summary
Determine how to ensure incorrect prop names are picked up by Angular's compiler - or Intellisense if using VSCode.

## 💬 Description
One comment from an Angular developer is that they tried to pass `appTitle` as a prop to the side navigation, when it should in fact be `app-title`. This was not caught by Angular's compiler or Intellisense so it was not obvious that it was incorrect until they found that the app title was not being rendered.

This ticket involves determining whether it is possible to ensure these sorts of issues are picked up and highlighted in an Angular developer's IDE, and then adding to the [Angular guidance](https://design.sis.gov.uk/get-started/install-components/angular) to instruct developers on how to enable it.

## 💰 Use value
This will allow errors with prop names to be picked up more easily and made clearer to developers building Angular apps.

**Comment by GCHQ-Developer-530** [comment]
I don't think this is an ICDS issue, after looking around it seems to be an ongoing complaint about Angular that misspelled prop names don't get picked up.

**Comment by GCHQ-Developer-398** [comment]
**Determine how incorrect prop names can be caught in Angular apps #499**

Intellisense is used to ensure incorrect prop names are picked up in VSCode. This issue is only resolvable if every developer downloads Intellisense to pick up on incorrect prop names. This issue is dependant on the developer, and it does not seem like there is anything we can do to pick up on this issue unless Intellisense is installed on every developers VSCode, which ultimately is the developers preference.

I don't think this is an ICDS issue, after looking around it seems to be an ongoing complaint about Angular. It does not pick up misspelled prop names.
Could write our own angular tests to check prop names? However this would be a very long and tedious process all for this small issue.

After this research I believe this is not an ICDS issue and so will close the issue.
